api/src/main/java/io/opentelemetry/api/baggage/ImmutableBaggage.java

```diff
@@ -146,14 +150,14 @@ public String toString() {
 
     @Override
     public Baggage.Builder put(String key, String value) {
-      entries.put(
-          Objects.requireNonNull(key, "key"),
-          Entry.create(key, Objects.requireNonNull(value, "value"), EntryMetadata.EMPTY));
-      return this;
+      return put(key, value, EntryMetadata.EMPTY);
     }
 
     @Override
     public Baggage.Builder remove(String key) {
+      if (key == null) {
+        return this;
+      }
       entries.remove(Objects.requireNonNull(key, "key"));
```

Comment:
This is wrong, either not allow and check for null or throw exception.

Reply:
Oops, removed the redundant checks